frontend/www/js/omegaup/components/course/StudentProgress.vue

```diff
@@ -120,6 +121,27 @@ export default class StudentProgress extends Vue {
     );
   }
 
+  get weightedMean(): Array<number> {
+    const totalPoints =
+      this.assignments
+        .map((assignment) => assignment.max_points)
+        .reduce((a, b) => (a ?? 0) + (b ?? 0)) ?? 0;
+    return this.assignments.map(
+      (assignment) => ((assignment.max_points ?? 0) / (totalPoints ?? 0)) * 100,
+    );
+  }
+
+  get globalScoreByStudent(): string {
+    const points = [];
+    for (const assignment of this.assignments) {
+      points.push(this.score(assignment.alias) / this.points(assignment.alias));
+    }
+    const globalScore = points
+      .map((value, index) => value * this.weightedMean[index])
+      .reduce((a, b) => a + b);
+    return globalScore.toFixed(2);
```

Comment:
```suggestion
    return points
      .map((value, index) => value * this.weightedMean[index])
      .reduce((a, b) => a + b, 0)
      .toFixed(2);
```

si algo se usa sólo una vez inmediatamente en la siguiente línea y no causa reducción de legibilidad darle inline, es mejor darle inline.

Reply:
Listo.